kosat-core/src/commonMain/kotlin/org/kosat/CDCL.kt

```diff
@@ -126,11 +125,9 @@ class CDCL {
 
         // Phase saving heuristics
         polarity.add(LBool.UNDEF)
-
-        return numberOfVariables
     }
 
-    fun value(lit: Lit): LBool {
+    private fun value(lit: Lit): LBool {
```

Comment:
This method is a part of public API

Reply:
Not anymore.

Reply:
But it should be :)
How else are you going to extract values of literals after solving the SAT? "Build a model" is not a right answer to this: (1) model might be too large (consider the situation when you really need the value of only one literal out of 1kk variables), (2) in the model you replace `LBool.Undef` with `true`, basically losing the information about the literal state.

Reply:
While it is possible to correctly return `LBool.Undef` right now, things might get interesting soon. 

> "Build a model" is not a right answer to this

Hilariously, "build a model" might be the right answer to this. I don't know the details of how this is implemented in MiniSat, but in Cadical, before every call to `External::ival`, it checks if the assignment have been "extended" (`External::extended`), and if not, it "extends" it. By "extension" authors mean "reconstruction". This operation is pretty much identical to building the model in our solver.

Consider BVE. The only way to get the value of eliminated variable is to traverse back the clauses which got resolved and check if those are satisfied. This is process takes O(length of all resolved clauses), worse case, and there is no way around it. It is true that "model might be too large", but reconstruction takes O(see above), a pretty trivial one, actually, while the solving takes however long it needs. The current implementation with substitution is potentially even faster (but I'm going to get rid of it soon). I think there is no reason in trying to optimize out reconstruction.

> in the model you replace LBool.Undef with true, basically losing the information about the literal state.

True, but this only triggers if the state of the solver is trivial (i.e. all clauses got simplified away). If we got to the search phase, we need the entire trail of assignments. I am willing to ignore that. From an API standpoint, I don't 
see a point of exposing values of partial assignments: those are private to the solver. Finally, in Cadical, `Solver::ival` also returns either `true` or `false` (kind of).